language/use-cases/translation_training_data/translation_training_data_tsv_generator.ipynb

```diff
@@ -0,0 +1,544 @@
+    "\n",
+    "It comes in [Basic and Advanced](https://cloud.google.com/translate/docs/editions) editions. Both provide fast and dynamic translation, but Advanced offers customization features, such as domain-specific translation, formatted document translation, and batch translation.\n",
+    "\n",
+    "The first 500,000 characters sent to the API to process (Basic and Advanced combined) per month are free (not applicable to LLMs).\n",
+    "\n",
+    "## Objective\n",
+    "\n",
+    "### Key Features\n",
+    "1. Paragraphs are converted into line-pairs of less than 200 words.\n",
+    "2. Tables in documents are converted into a line-pair with each row as a separate line-pair.\n",
+    "3. Limit of 200 words per line is handled.\n",
+    "4. Empty or blank lines are not added to the TSV.\n",
+    "\n",
+    "This notebook enables you to generate a TSV file out of documents(docx) for training NMT(neural machine translation) model. The generated TSV file will contain the source and target line pairs for 2 languages in 2 columns respectively. Limit of 200 words for a line is handled within the code. Example: If a line is exceeding 200 words, it won't be added to the training dataset, but will be captured and returned in a dictionary so that you can decide on how to convert it to line-pair of less than 200 words.\n",
+    "The code also removes any blank or empty lines in a document from both source and reference before making line-pairs. This makes sure that both the documents do not mismatch with line-pairs due to empty lines.\n",
+    "\n",
+    "\n",
+    "## How to use the notebook\n",
+    "\n",
+    "##### input: a dictionary containing source and reference GCS paths.\n",
+    "\n",
+    "##### output: a single TSV file, 2 dictionaries\n",
+    "\n",
+    "##### Steps to follow:\n",
+    "- Provide as many source and reference files in the input dictionary: `source_ref_dictionary`, *key* being the source file path and reference file path as its *value*\n",
+    "- Trigger all the cells after providing the input.\n",
+    "- The TSV gets created in your local path.\n",
+    "\n",
+    "\n",
+    "\n",
+    "## Costs\n",
+    "\n",
+    "Learn about [Translation pricing](https://cloud.google.com/translate/pricing) and use the [Pricing Calculator](https://cloud.google.com/products/calculator/) to generate a cost estimate based on your projected usage."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e093fb72-bc78-49c0-95e3-01ac8b539da7",
+   "metadata": {},
+   "source": [
+    "## **Getting Started**\n",
+    "### Install docx and pyPDF2 SDK for Python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "99abf771-97ee-4d53-8a90-eb147ecfa0c8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# !pip install --proxy \"\" pyPDF2 --quiet\n",
+    "# !pip install --proxy \"\" docx --quiet"
+   ]
```

Comment:
I would prefer if this notebook used something like Google Cloud Document AI to parse the DOCX or PDF file, showcases more Google Cloud technology.

Reply:
Ok, that's fine